statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+            Mean bootstrap averages the number of resampled clusters over bsn
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        clust_count = np.zeros(self.design.nclust)
+        w = self.weights.copy()
+        # for each replicate, I accumulate bsn number of times?
+        for b in range(bsn):
+            for s in range(self.nstrat):
+                w[self.strat == s] *= ((float(self.ncs[s] - 1) /
+                                        self.ncs[s])**(1/bsn))
+                # If there is only one or two clusters then weights wont change
+                if (self.ncs[s] == 1 or self.ncs[s] == 2):
+                    continue
+                # resample array of clusters in strata s
+                ii_resample = np.random.choice(self.ii[s], size=(self.ncs[s]-1))
+                # accumulate number of times cluster i was resampled
+                clust_count += np.bincount(ii_resample,
+                                           minlength=max(self.sclust)+1)
+        # avg number of times cluster i was resampled
+        clust_count /= bsn
+        # augment weights
+        w *= clust_count[self.sclust]
+        return w
+
+
+class SurveyStat(SurveyDesign):
+    """
+    Estimation and inference for summary statistics in complex surveys.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+
+    Attributes
+    ----------
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+
+    def __init__(self, design, mse=False):
+        self.design = design
+        self.mse = mse
+
+    def _bootstrap(self, replicates=None, bsn=None):
+        """
+        Calculates bootstrap standard errors
+
+        Parameters
+        ----------
+        stat : object
+            Object of class SurveyMean, SurveyTotal, SurveyPercentile, etc
+        replicates : integer
+            The number of replicates that the user wishes to specify
+
+        Returns
+        -------
+        est : ndarray
+            The point estimates of the statistic, calculated on the columns
+            of data.
+        vcov : ndarray
+            The variance-covariance of the estimates.
+        """
+        est = self._stat(self.design.weights)
+
+        jdata = []
+        self.w_li = []
+        for i in range(replicates):
+            w = self.design.get_rep_weights(i, bsn=bsn)
+            self.w_li.append(w)
+            jdata.append(self._stat(w))
+        jdata = np.asarray(jdata)
+        if self.mse:
+            print("mse specified")
+            var = np.dot((jdata-est).T, (jdata-est)) / replicates
+            var = np.sqrt(np.diag(var))
+        else:
+            boot_mean = jdata.mean(0)
+            var = ((jdata - boot_mean)**2).sum(0) / replicates
+
+        return est, np.sqrt(var)
```

Comment:
Should return vcov not se.  Also should work for multiple vairables so `np.sqrt(var)` would not work when `var` is a matrix.